commit 2aac0a862334f3370a7e0b6bc7a53cf0ab847182
Author: Siddhesh Poyarekar <siddhesh@redhat.com>
Date:   Thu Jun 26 14:10:07 2014 +0530

    Fix Wundef warning for SEPARATE_KEY
    
    The SEPARATE_KEY macro is never defined in any condition or
    architecture.  It is effectively dead code, so remove it.

diff --git a/nscd/connections.c b/nscd/connections.c
index 57cb8c180e8e6402..53813cf58876eae7 100644
--- a/nscd/connections.c
+++ b/nscd/connections.c
@@ -317,12 +317,6 @@ enum usekey
     use_he = 1,
     use_he_begin = use_he | use_begin,
     use_he_end = use_he | use_end,
-#if SEPARATE_KEY
-    use_key = 2,
-    use_key_begin = use_key | use_begin,
-    use_key_end = use_key | use_end,
-    use_key_first = use_key_begin | use_first,
-#endif
     use_data = 3,
     use_data_begin = use_data | use_begin,
     use_data_end = use_data | use_end,
@@ -471,16 +465,7 @@ verify_persistent_db (void *mem, struct database_pers_head *readhead, int dbnr)
 	  if (here->key < here->packet + sizeof (struct datahead)
 	      || here->key > here->packet + dh->allocsize
 	      || here->key + here->len > here->packet + dh->allocsize)
-	    {
-#if SEPARATE_KEY
-	      /* If keys can appear outside of data, this should be done
-		 instead.  But gc doesn't mark the data in that case.  */
-	      if (! check_use (data, head->first_free, usemap,
-			       use_key | (here->first ? use_first : 0),
-			       here->key, here->len))
-#endif
-		goto fail;
-	    }
+	    goto fail;
 
 	  work = here->next;
 
@@ -500,10 +485,6 @@ verify_persistent_db (void *mem, struct database_pers_head *readhead, int dbnr)
      he->first == true hashentry.  */
   for (ref_t idx = 0; idx < head->first_free; ++idx)
     {
-#if SEPARATE_KEY
-      if (usemap[idx] == use_key_begin)
-	goto fail;
-#endif
       if (usemap[idx] == use_data_begin)
 	goto fail;
     }
